Use normal constructor instead of record constructor for i18n messages

L0neGamer opened the pull request:
fixes #289

parsonsmatt commented:
This is definitely a breaking change, at least. I'm not sure if this is something we want to do - I don't personally use these and so I don't know what the potential impact will be. 

Alternatively, we can introduce another setting that controls if it is record or regular constructors, which is record by default. That way it's not a breaking change, and you get the behavior you want.

L0neGamer commented:
@parsonsmatt I've introduced an options type like I've done previously, exposing setters but not getters nor the internals of the structure so that we can freely manipulate its internals.

I've also extended testing for i18n slightly to test that both cases work.

It also seems there may have been a bug where the constructor prefix wasn't set properly when `toCon` was used internally. I've fixed this too.

L0neGamer commented:
@jezen if you wouldn't mind
